xbmc/input/JoystickMapper.cpp

```diff
@@ -38,6 +38,15 @@ void CJoystickMapper::MapActions(int windowID, const TiXmlNode *pDevice)
   if (controllerId.empty())
     return;
 
+  // Update Controller IDs
```

Comment:
Is this block of code still needed?

Reply:
I think this block is the one that's actually needed. But I think the other code (in the if-statement) can be removed. I will do that.